Firestore/core/src/firebase/firestore/auth/firebase_credentials_provider_apple.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Right now, FirebaseCredentialsProvider only support APPLE build.
+#if !defined(__OBJC__)
+#error "This header only supports Objective-C++."
+#endif  // !defined(__OBJC__)
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_AUTH_FIREBASE_CREDENTIALS_PROVIDER_APPLE_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_AUTH_FIREBASE_CREDENTIALS_PROVIDER_APPLE_H_
+
+#import <Foundation/Foundation.h>
+
+#include <memory>
+#include <mutex>  // NOLINT(build/c++11)
+
+#include "Firestore/core/src/firebase/firestore/auth/credentials_provider.h"
+#include "Firestore/core/src/firebase/firestore/auth/user.h"
+#include "absl/strings/string_view.h"
+
+@class FIRApp;
+
+namespace firebase {
+namespace firestore {
+namespace auth {
+
+/**
+ * `FirebaseCredentialsProvider` uses Firebase Auth via `FIRApp` to get an auth
+ * token.
+ *
+ * NOTE: To simplify the implementation, it requires that you set
+ * `userChangeListener` with a non-`nil` value no more than once and don't call
+ * `getTokenForcingRefresh:` after setting it to `nil`.
+ *
+ * This class must be implemented in a thread-safe manner since it is accessed
+ * from the thread backing our internal worker queue and the callbacks from
+ * FIRAuth will be executed on an arbitrary different thread.
+ *
+ * For non-Apple desktop build, this is right now just a stub.
+ */
+class FirebaseCredentialsProvider : public CredentialsProvider {
+ public:
+  // TODO(zxu123): Provide a ctor to accept the C++ Firebase Games App, which
+  // deals all platforms. Right now, only works for FIRApp*.
+  /**
+   * Initializes a new FirebaseCredentialsProvider.
+   *
+   * @param app The Firebase app from which to get credentials.
+   */
+  explicit FirebaseCredentialsProvider(FIRApp* app);
+
+  void GetToken(bool force_refresh, TokenListener completion) override;
+
+  void SetUserChangeListener(UserChangeListener listener) override;
+
+  friend class FirebaseCredentialsProviderTest_GetToken_Test;
+  friend class FirebaseCredentialsProviderTest_SetListener_Test;
+  friend class DISABLED_FirebaseCredentialsProviderTest_GetToken_Test;
+  friend class DISABLED_FirebaseCredentialsProviderTest_SetListener_Test;
+
+ private:
+  /** Initialize with default app for internal usage such as test. */
+  FirebaseCredentialsProvider();
+
+  static void PlatformDependentTestSetup(const absl::string_view config_path);
```

Comment:
`PlatformDependentTestSetup` appears to no longer have a definition. Can this be removed?